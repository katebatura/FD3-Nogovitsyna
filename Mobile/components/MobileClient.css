td {
  border-top: 1px solid gray;
  padding: 10px;
}

.active {
  background-color: green;
}
.blocked {
  background-color: red;
}
