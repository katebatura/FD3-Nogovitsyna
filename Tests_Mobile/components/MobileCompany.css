.MobileCompany
{
  border: 1px solid gray;
  border-collapse: collapse;
  width: 1000px;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  font-size: 15px;
  text-align: center;
  vertical-align: center;
  padding: 10px;
  margin: 20px 0;
  
}

.MobileCompanyName
{
  font-weight: bold;
}

th {
  border-top: 1px solid gray;
  padding: 10px;
}

.watchClients {
  background-color: darkgray;
}
