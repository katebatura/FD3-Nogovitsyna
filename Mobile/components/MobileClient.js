import React from 'react';
import PropTypes from 'prop-types';

import './MobileClient.css';
import {events} from './events';

class MobileClient extends React.PureComponent {

  static propTypes = {
    info: PropTypes.shape({
      id: PropTypes.number.isRequired,
      fam: PropTypes.string.isRequired,
      im: PropTypes.string.isRequired,
      otch: PropTypes.string.isRequired,
      balance: PropTypes.number.isRequired,
    }),
  };

  state = {
    info: this.props.info,
  };

  componentWillReceiveProps = (newProps) => {
    console.log("MobileClient id=" + this.props.info.id + " componentWillReceiveProps");
    this.setState( {info: newProps.info} );
  };

  editClient = () => {
    events.emit('editClient', this.state.info.id);
  };

  deleteClient = ()=> {
    events.emit('deleteClient', this.state.info.id);
  };

  render() {

    console.log("MobileClient id=" + this.state.info.id + " render");
    
    return (
      <tr>
        <td>{this.state.info.fam}</td>
        <td>{this.state.info.im}</td>
        <td>{this.state.info.otch}</td>
        <td>{this.state.info.balance}</td>
        {
          (this.state.info.balance >= 0)?
          <td className = "active">Active</td>:
          <td className = "blocked">Blocked</td>
        }
        <td>
          <input type = "button" value = "Редактировать" onClick = {this.editClient} />          
        </td>
        <td>
          <input type = "button" value = "Удалить" onClick = {this.deleteClient} />          
        </td>
      </tr>
    );

  }

}

export default MobileClient;
