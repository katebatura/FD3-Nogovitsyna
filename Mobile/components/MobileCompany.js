import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';
import Card from './card';
import {events} from './events';

import './MobileCompany.css';


class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
        otch: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    clients: this.props.clients,
    reservedClients: this.props.clients,
    workMode: 0,
    selectedClient: 0,
    new: false,
    watchClients: 0, //1-active Clients, 2- blocked Clients, 0 - all Clients
  };

  setName1 = () => {
    this.setState( {name:'МТС'} );
  };

  setName2 = () => {
    this.setState( {name:'Velcom'} );
  };

  setActiveClients = () => {
    let activeClients = this.state.reservedClients.filter(v => {return v.balance >= 0});
    this.setState( {clients: activeClients, watchClients: 1} );
  };

  setBlockedClients = () => {
    let blockedClients = this.state.reservedClients.filter(v => {return v.balance < 0});
    this.setState( {clients: blockedClients, watchClients: 2} );
  };

  setAllClients = () => {
    let allClients = [...this.state.reservedClients];
    this.setState( {clients: allClients, watchClients: 0} );
  };

  addNewClient = () => {
    if(this.state.workMode != 2){
      let newId = this.state.reservedClients[this.state.reservedClients.length - 1].id + 1;
      this.setState( {workMode: 2, selectedClient: newId, new: true} );
    } else {
      this.setState( {workMode: 0} );
    }    
  };

  componentDidMount = () => {
    events.addListener('saveClient',this.saveClient);
    events.addListener('cancelChanges',this.cancelChanges);
    events.addListener('editClient',this.editClient);
    events.addListener('deleteClient',this.deleteClient);
  };

  componentWillUnmount = () => {
    events.removeListener('saveClient',this.saveClient);
    events.removeListener('cancelChanges',this.cancelChanges);
    events.removeListener('deleteClient',this.deleteClient);
    events.removeListener('editClient',this.editClient);
  } ;
  
  editClient = (client) => {
    this.setState( {workMode: 1, selectedClient: client, new: false} );
  };

  deleteClient = (client) => {
    let newReservedClients = this.state.reservedClients.filter(c => {
      return c.id != client
    });
    let newClients = this.state.clients.filter(c => {
      return c.id != client
    });
    this.setState( {clients: newClients, reservedClients: newReservedClients,
       workMode: this.state.selectedClient == client ? 0 : this.state.workMode} );
  };

  saveClient = (client) => {
    this.setState( {workMode: null} );
    if(this.state.new) {
      let newReservedClients = [...this.state.reservedClients, client];
      this.setState( {reservedClients: newReservedClients, new: false} );

      switch(this.state.watchClients) {
        case 0:
          this.setState( {clients: newReservedClients} )
          break;
        case 1:
          if(client.balance >= 0){
            let newClients = [...this.state.clients, client]
            this.setState( {clients: newClients} )
          }
          break;
        case 2:
        if(client.balance < 0){
          let newClients = [...this.state.clients, client]
          this.setState( {clients: newClients} )
        }
      }
      return
    } 
    let changed = false;
    let newClients = [...this.state.clients]; // копия самого массива клиентов
    newClients.forEach( (c,i) => {
      if ( c.id == client.id ) {
        let newClient = {...c}; // копия хэша изменившегося клиента
        newClient.fam = client.fam;
        newClient.im = client.im;
        newClient.otch = client.otch;
        newClient.balance = client.balance;
        newClients[i] = newClient;
        changed = true;
      }
    } );
    switch(this.state.watchClients){
      case 1:
        newClients = newClients.filter(v => {return v.balance >= 0})
        break;
      case 2:
        newClients = newClients.filter(v => {return v.balance < 0})
        break;
    }

    if ( changed ){
      let newReservedClients = [...this.state.reservedClients]; // копия самого массива клиентов
      newReservedClients.forEach( (c,i) => {
        if ( c.id == client.id ) {
          let newClient = {...c}; // копия хэша изменившегося клиента
          newClient.fam = client.fam;
          newClient.im = client.im;
          newClient.otch = client.otch;
          newClient.balance = client.balance;
          newReservedClients[i] = newClient;
        }
      } );
      this.setState( {clients: newClients, reservedClients: newReservedClients,} );
    }
  };

  cancelChanges = () => {
    this.setState( {workMode: null, selectedClient: null} );
  };
    
  render() {

    console.log("MobileCompany render");
    var clientsCode = this.state.clients.map( client => {         
      return <MobileClient key = {client.id} info = {client} />;              
      }
    );

    return (
      <div>
        <input type = "button" value = "МТС" onClick = {this.setName1} />
        <input type = "button" value = "Velcom" onClick = {this.setName2} />
        <div className = 'MobileCompanyName'> Компания &laquo;{this.state.name}&raquo; </div>
        <hr />
        <input type = "button" value = "Все" onClick = {this.setAllClients} className = {!this.state.watchClients ? "watchClients" : null} />
        <input type = "button" value = "Активные" onClick = {this.setActiveClients} className = {this.state.watchClients == 1 ? "watchClients" : null} />
        <input type = "button" value = "Заблокированные" onClick = {this.setBlockedClients} className = {this.state.watchClients == 2 ? "watchClients" : null} />
        <hr />
        <table className = 'MobileCompany'>
          <tbody>
            <tr>
              <th>Фамилия</th>
              <th>Имя</th>
              <th>Отчество</th>
              <th>Баланс</th>
              <th>Статус</th>
              <th>Редактировать</th>
              <th>Удалить</th>
            </tr>
            {clientsCode}
          </tbody>
        </table>
        <input type = "button" value = "Добавить клиента" onClick = {this.addNewClient} />
        {
          this.state.workMode == 1 ? 
          <Card
          key = {this.state.selectedClient}
          workMode = {this.state.workMode}
          selectedClient = {this.state.selectedClient}
          info = {this.state.selectedClient ? this.state.clients.filter( client => {
           if(client.id == this.state.selectedClient) {
            return client
           }
          })[0]: null } >
          </Card> : null
        }
        {
          this.state.workMode == 2 ? 
          <Card
          key = {this.state.selectedClient}
          workMode = {this.state.workMode}
          selectedClient = {this.state.selectedClient}
          info = { {id: this.state.selectedClient, key: this.state.selectedClient, fam: '',
          im: '', otch: '', balance: 0 } } >
          </Card> : null
        }
      </div>
    );

  }

}

export default MobileCompany;
