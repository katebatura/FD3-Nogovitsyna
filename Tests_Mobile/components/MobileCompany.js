import React from 'react';
import PropTypes from 'prop-types';

import MobileClient from './MobileClient';
import Card from './card';
import ChangeClients from './clients';
import {events} from './events';

import './MobileCompany.css';


class MobileCompany extends React.PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    clients: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        fam: PropTypes.string.isRequired,
        im: PropTypes.string.isRequired,
        otch: PropTypes.string.isRequired,
        balance: PropTypes.number.isRequired,
      })
    ),
  };

  state = {
    name: this.props.name,
    clients: this.props.clients,
    reservedClients: this.props.clients,
    workMode: 0,
    selectedClient: 0,
    new: false,
    watchClients: 0, //1-active Clients, 2- blocked Clients, 0 - all Clients
  };

  setName1 = () => {
    this.setState( {name:'МТС'} );
  };

  setName2 = () => {
    this.setState( {name:'Velcom'} );
  };

  setActiveClients = () => {

    let activeClients = ChangeClients.setActiveClients(this.state.reservedClients);

    this.setState( {clients: activeClients, watchClients: 1} );

  };

  setBlockedClients = () => {

    let blockedClients = ChangeClients.setBlockedClients(this.state.reservedClients);

    this.setState( {clients: blockedClients, watchClients: 2} );

  };

  setAllClients = () => {

    let allClients = ChangeClients.setAllClients(this.state.reservedClients);

    this.setState( {clients: allClients, watchClients: 0} );

  };

  addNewClient = () => {

    if(this.state.workMode != 2){
      let newId = ChangeClients.addNewClient(this.state.reservedClients);

      this.setState( {workMode: 2, selectedClient: newId, new: true} );

    } else {
      this.setState( {workMode: 0} );
    }  

  };

  componentDidMount = () => {
    events.addListener('saveClient',this.saveClient);
    events.addListener('cancelChanges',this.cancelChanges);
    events.addListener('editClient',this.editClient);
    events.addListener('deleteClient',this.deleteClient);
  };

  componentWillUnmount = () => {
    events.removeListener('saveClient',this.saveClient);
    events.removeListener('cancelChanges',this.cancelChanges);
    events.removeListener('deleteClient',this.deleteClient);
    events.removeListener('editClient',this.editClient);
  } ;
  
  editClient = (client) => {
    
    this.setState( {workMode: 1, selectedClient: client, new: false} );

  };

  deleteClient = (client) => {

    let newState = ChangeClients.deleteClient( client, this.state.clients, this.state.reservedClients);

    this.setState( {clients: newState.newClients, reservedClients: newState.newReservedClients,
       workMode: this.state.selectedClient == client ? 0 : this.state.workMode} );

  };

  saveClient = (client) => {

    this.setState( {workMode: null} );

    if(this.state.new) {
      let newState = ChangeClients.saveNewClient( client, this.state.clients, this.state.reservedClients);

      this.setState( {reservedClients: newState.newReservedClients, new: false} );

      switch(this.state.watchClients) {
        case 0:
          this.setState( {clients: newState.newReservedClients} )
          break;
        case 1:
          if(client.balance >= 0){
            this.setState( {clients: newState.newClients} )
          }
          break;
        case 2:
        if(client.balance < 0){
          this.setState( {clients: newState.newClients} )
        }
      }
      return
    };

    
    let newState = ChangeClients.saveEditClient( client, this.state.clients, this.state.reservedClients);

    let newClients = newState.newClients;

    switch(this.state.watchClients){
      case 1:
        newClients = newState.newActiveClients
        break;
      case 2:
        newClients = newState.newBlockedClients
        break;
    }

    if ( newState.changed ){      
      this.setState( {clients: newClients, reservedClients: newState.newReservedClients,} );
    }
  };

  cancelChanges = () => {
    this.setState( {workMode: null, selectedClient: null} );
  };
    
  render() {

    console.log("MobileCompany render");
    var clientsCode = this.state.clients.map( client => {         
      return <MobileClient key = {client.id} info = {client} />;              
      }
    );

    return (
      <div>
        <input type = "button" value = "МТС" onClick = {this.setName1} />
        <input type = "button" value = "Velcom" onClick = {this.setName2} />
        <div className = 'MobileCompanyName'> Компания &laquo;{this.state.name}&raquo; </div>
        <hr />
        <input type = "button" value = "Все" onClick = {this.setAllClients} className = {!this.state.watchClients ? "watchClients" : null} />
        <input type = "button" value = "Активные" onClick = {this.setActiveClients} className = {this.state.watchClients == 1 ? "watchClients" : null} />
        <input type = "button" value = "Заблокированные" onClick = {this.setBlockedClients} className = {this.state.watchClients == 2 ? "watchClients" : null} />
        <hr />
        <table className = 'MobileCompany'>
          <tbody>
            <tr>
              <th>Фамилия</th>
              <th>Имя</th>
              <th>Отчество</th>
              <th>Баланс</th>
              <th>Статус</th>
              <th>Редактировать</th>
              <th>Удалить</th>
            </tr>
            {clientsCode}
          </tbody>
        </table>
        <input type = "button" value = "Добавить клиента" onClick = {this.addNewClient} id = "addClientButton"  />
        {
          this.state.workMode == 1 ? 
          <Card
          key = {this.state.selectedClient}
          workMode = {this.state.workMode}
          selectedClient = {this.state.selectedClient}
          info = {this.state.selectedClient ? this.state.clients.filter( client => {
           if(client.id == this.state.selectedClient) {
            return client
           }
          })[0]: null } >
          </Card> : null
        }
        {
          this.state.workMode == 2 ? 
          <Card
          key = {this.state.selectedClient}
          workMode = {this.state.workMode}
          selectedClient = {this.state.selectedClient}
          info = { {id: this.state.selectedClient, key: this.state.selectedClient, fam: '',
          im: '', otch: '', balance: 0 } } >
          </Card> : null
        }
      </div>
    );

  }

}

export default MobileCompany;
