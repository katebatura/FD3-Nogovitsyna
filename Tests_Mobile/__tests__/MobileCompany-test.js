"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileCompany from '../components/MobileCompany';
import MobileClient from '../components/MobileClient';

test('работа кнопок "Добавить", "Редактировать" и "Удалить" клиента', () => {
    
    //исходные данные
    let companyName='Velcom';
    let clientsArr=[ 
    {id:1, fam:"Иванов", im:"Иван", otch:"Иванович", balance:200}, 
    {id:2, fam:"Сидоров", im:"Сидор", otch:"Сидорович", balance:250}, 
    {id:3, fam:"Петров", im:"Пётр", otch:"Петрович", balance:180},
    {id:4, fam:"Григорьев", im:"Григорий", otch:"Григорьевич", balance:-220},
    ];

    // создаём тестовую версию компонента MobileCompany
    const component  = renderer.create(
        <MobileCompany
        name={companyName}
        clients={clientsArr}
       />
    );
    // получаем снэпшот MobileCompany (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // найдём в вёрстке компонента кнопку addClientButton
    const addClientButton = component.root.find( el => el.props.id == 'addClientButton'  ); 
    addClientButton.props.onClick();

    // получаем уже изменённый снэпшот MobileCompany
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    // "нажмём" кнопку ещё раз
    addClientButton.props.onClick();
    
    // и получаем окончательный снэпшот MobileCompany
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();



    //создаём тестовую версию компонента MobileClient
    const client = renderer.create(
        <MobileClient
        info = {clientsArr[0]}/>
    )

    // найдём в вёрстке компонента кнопку editClientButton
    const editClientButton = client.root.find( el => el.props.id == 'editClientButton' ); 
    editClientButton.props.onClick();

    // получаем уже изменённый снэпшот  MobileCompany
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    
    // найдём в вёрстке компонента кнопку deleteClientButton
    const deleteClientButton = client.root.find( el => el.props.id == 'deleteClientButton' ); 
    deleteClientButton.props.onClick();

    // получаем уже изменённый снэпшот  MobileCompany
    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
});