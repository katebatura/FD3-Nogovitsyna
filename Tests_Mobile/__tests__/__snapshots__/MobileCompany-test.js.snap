// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа кнопок "Добавить", "Редактировать" и "Удалить" клиента 1`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="МТС"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <div
    className="MobileCompanyName"
  >
     Компания «
    Velcom
    » 
  </div>
  <hr />
  <input
    className="watchClients"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompany"
  >
    <tbody>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          Blocked
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="addClientButton"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок "Добавить", "Редактировать" и "Удалить" клиента 2`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="МТС"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <div
    className="MobileCompanyName"
  >
     Компания «
    Velcom
    » 
  </div>
  <hr />
  <input
    className="watchClients"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompany"
  >
    <tbody>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          Blocked
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="addClientButton"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <div>
    <h3>
       New Client 
    </h3>
    <label>
      Фамилия: 
      <input
        defaultValue=""
      />
       
    </label>
    <br />
    <label>
      Имя: 
      <input
        defaultValue=""
      />
       
    </label>
    <br />
    <label>
      Отчество: 
      <input
        defaultValue=""
      />
       
    </label>
    <br />
    <label>
      Баланс: 
      <input
        defaultValue={0}
      />
       
    </label>
    <br />
    <button
      onClick={[Function]}
    >
      Save
    </button>
    <button
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`работа кнопок "Добавить", "Редактировать" и "Удалить" клиента 3`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="МТС"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <div
    className="MobileCompanyName"
  >
     Компания «
    Velcom
    » 
  </div>
  <hr />
  <input
    className="watchClients"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompany"
  >
    <tbody>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          Blocked
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="addClientButton"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;

exports[`работа кнопок "Добавить", "Редактировать" и "Удалить" клиента 4`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="МТС"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <div
    className="MobileCompanyName"
  >
     Компания «
    Velcom
    » 
  </div>
  <hr />
  <input
    className="watchClients"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompany"
  >
    <tbody>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
      <tr>
        <td>
          Иванов
        </td>
        <td>
          Иван
        </td>
        <td>
          Иванович
        </td>
        <td>
          200
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          Blocked
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="addClientButton"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
  <div>
    <h3>
       Edit Client 
    </h3>
    <label>
      Фамилия: 
      <input
        defaultValue="Иванов"
      />
       
    </label>
    <br />
    <label>
      Имя: 
      <input
        defaultValue="Иван"
      />
       
    </label>
    <br />
    <label>
      Отчество: 
      <input
        defaultValue="Иванович"
      />
       
    </label>
    <br />
    <label>
      Баланс: 
      <input
        defaultValue={200}
      />
       
    </label>
    <br />
    <button
      onClick={[Function]}
    >
      Save
    </button>
    <button
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;

exports[`работа кнопок "Добавить", "Редактировать" и "Удалить" клиента 5`] = `
<div>
  <input
    onClick={[Function]}
    type="button"
    value="МТС"
  />
  <input
    onClick={[Function]}
    type="button"
    value="Velcom"
  />
  <div
    className="MobileCompanyName"
  >
     Компания «
    Velcom
    » 
  </div>
  <hr />
  <input
    className="watchClients"
    onClick={[Function]}
    type="button"
    value="Все"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Активные"
  />
  <input
    className={null}
    onClick={[Function]}
    type="button"
    value="Заблокированные"
  />
  <hr />
  <table
    className="MobileCompany"
  >
    <tbody>
      <tr>
        <th>
          Фамилия
        </th>
        <th>
          Имя
        </th>
        <th>
          Отчество
        </th>
        <th>
          Баланс
        </th>
        <th>
          Статус
        </th>
        <th>
          Редактировать
        </th>
        <th>
          Удалить
        </th>
      </tr>
      <tr>
        <td>
          Сидоров
        </td>
        <td>
          Сидор
        </td>
        <td>
          Сидорович
        </td>
        <td>
          250
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Петров
        </td>
        <td>
          Пётр
        </td>
        <td>
          Петрович
        </td>
        <td>
          180
        </td>
        <td
          className="active"
        >
          Active
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
      <tr>
        <td>
          Григорьев
        </td>
        <td>
          Григорий
        </td>
        <td>
          Григорьевич
        </td>
        <td>
          -220
        </td>
        <td
          className="blocked"
        >
          Blocked
        </td>
        <td>
          <input
            id="editClientButton"
            onClick={[Function]}
            type="button"
            value="Редактировать"
          />
        </td>
        <td>
          <input
            id="deleteClientButton"
            onClick={[Function]}
            type="button"
            value="Удалить"
          />
        </td>
      </tr>
    </tbody>
  </table>
  <input
    id="addClientButton"
    onClick={[Function]}
    type="button"
    value="Добавить клиента"
  />
</div>
`;
